vulnerabilities/package_managers.py

```diff
@@ -347,22 +429,51 @@ class GitHubTagsAPI(VersionAPI):
     package_type = "github"
 
     async def load_api(self, repo_set):
-        async with client_session() as session:
+        session = client_session()
+        async with session as session:
             await asyncio.gather(
                 *[
-                    self.fetch(owner_repo.lower(), session)
+                    self.fetch(owner_repo.lower())
                     for owner_repo in repo_set
                     if owner_repo.lower() not in self.cache
                 ]
             )
 
-    async def fetch(self, owner_repo: str, session) -> None:
+    async def fetch(self, owner_repo: str, endpoint=None) -> None:
         # owner_repo is a string of format "{repo_owner}/{repo_name}"
         # Example value of owner_repo = "nexB/scancode-toolkit"
-        endpoint = f"https://api.github.com/repos/{owner_repo}/git/refs/tags"
-        resp = await session.request(method="GET", url=endpoint)
-        resp = await resp.json()
-        self.cache[owner_repo] = [release["ref"].split("/")[-1] for release in resp]
+        if owner_repo not in self.cache:
+            self.cache[owner_repo] = set()
+
+        if not endpoint:
+            endpoint = f"https://github.com/{owner_repo}/tags"
+        async with client_session() as session:
+            resp = await session.get(endpoint)
+            resp = await resp.read()
+
+        soup = BeautifulSoup(resp, features="lxml")
+        for release_entry in soup.find_all("div", {"class": "commit"}):
+            version = None
+            for links in release_entry.find_all("a"):
+                if f"/{owner_repo}/releases/tag/" in links["href"].lower():
+                    version = links["href"].split("/")[-1]
```

Comment:
always prefer `partition` over split for robustness an clarity. For instance
```suggestion
                    _prefix, _slash, version = links["href"].rpartition("/")
```